src/app/clusters/window-covering-server/window-covering-delegate.h

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandResponseHelper.h>
+#include <app/util/af.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace WindowCovering {
+
+/** @brief
+ *    Defines methods for implementing application-specific logic for the WindowCovering Cluster.
+ */
+class Delegate
+{
+public:
+    virtual CHIP_ERROR HandleLiftMovement() = 0;
```

Comment:
These `Handle` need some documentation: what is one supposed to do when `HandleFOO`? when is that method called and what is the expected operation? What does it mean for them to return errors?